execution_workflows/APA-Scan/workflow/Snakefile

```diff
@@ -0,0 +1,357 @@
+#       are present and complete. This is so that you can ensure that it is
+#       updated for new files/sets of samples etc.
+
+
+### RULE GROUPING
+# I've chosen to group all the rules together as the only one that requires any
+# real computational power is run_apa_scan. The rest are literally just file-
+# editing, so it's greedy and wasteful that they get their own cores. This might
+# mean you don't actually need --forcerun especially on a HPC/cloud executor,
+# but I'm not sure and it's best to be safe I think.
+
+
+### CODING
+# All I get when running snakemake --lint is it complaining that the text of
+# shell section of the rule generate_configurationfile refers to an absolute
+# path, however this is actually not a huge issue because I'm using
+# "os.getcwd()" meaning that the script will be tailored to the path of the
+# machine it's been run on. As far as I can tell you NEED to provide the abs-
+# olute path in the configuration.ini file else APA-Scan.py gets aggy and starts
+# throwing out errors.
+# I also get an error saying that I'm using old string formatting and should be
+# using "f". I actually find the string addition format really easy to read and
+# I think for this purpose it actually works well. I also don't see it really
+# causing any issues to the actual workflow itself.
+
+
+
+
+
+# ------------------------------------------------------------------------------
+# Packages, Libraries and Sources
+
+import pandas as pd
+
+
+
+
+
+# ------------------------------------------------------------------------------
+# Config/Metadata
+
+# Don't actually need this for the samplenames, you just need to know which
+# samples belong to which condition (ie in the test_data, control or srsf3).
+# I'm tempted to modify the APA-Scan.py script to work more like a standard
+# command line function, but obvs that'll depend on time
+
+
+# Read in config file
+configfile: "config/config.APA-Scan.yaml"
+
+# Convert config to dataframe
+samples = pd.read_csv(os.path.abspath(config["samples"]),sep=",")
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Parameters, filenames & sample data
+
+
+#### NOTES #####################################################################
+
+# APA-Scan requires your 2 levels of input bams to be in their own separate folders.
+# So I imagine this'll carry on from the previous RNA-Seq alignment step. They just
+# need to be in bam format and there can be multiple in a directory.
+# MAY NEED TO ADJUST THE INDEXING DEPENDING ON THE LEVELS, BUT CAN PROBABLY MAKE
+# THIS CUSTOMISABLE FROM THE COMMAND LINE OR A CONFIG FILE ETC. THIS IS BECAUSE
+# THE LAYOUT OF THESE WILL DICTATE THE ORIENTATION OF LEVELS THAT ARE COMPARED
+# BY APA-SCAN. IE FOR THE TEST DATA, THIS MAKES SO SRSF3 is CONDITION1 AND
+# THE CONTROL IS CONDITION 2
+
+################################################################################
+
+
+
+# Parameters
+THREADS = 8
```

Comment:
same as below... better define in config.
Ideally the whole workflow could be run with different samples, annotations, outputnames, threads, ...., anything that comes to mind, WITHOUT changing the code anywhere.